server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsRequest.java

```diff
@@ -331,7 +331,7 @@ public void writeTo(StreamOutput out) throws IOException {
         SCRIPT("script"),
         DISCOVERY("discovery"),
         INGEST("ingest"),
-        ADAPTIVE_SELECTION("adaptiveSelection");
+        ADAPTIVE_SELECTION("adaptive_selection");
```

Comment:
I'm not sure changing this will work in 7.x, because in mixed cluster, the server/request would not align? Additionally, the transport client would have similar failure scenarios?

Reply:
My plan is that this PR will change the token in `7.7`, which will mean that `7.7.0` and all following `7.7.x` releases will have the change. #53637 will make the change in `master` and the change will come in for `7.x` in #54141. Both of those PRs are ready for merge once this issue is resolved. I believe that if we make this change on `7.7`, `7.x`, and `master`, then there won't be any releases where the versions won't align. Does that make sense? Am I missing something?

Reply:
Thanks for the explanation. I just realized where my confusion lies: I was thinking this enum existed and these strings were serialized currently, when in fact the 7.6 and before request is serialized with boolean flags.